_posts/2025-02-27-use-desctop-visual-studio-code.adoc

```diff
@@ -0,0 +1,108 @@
+
+== Connecting Visual Studio Code desktop to a new workspace
+
+You can connect your local Visual Studio Code to a new workpace with a few clicks.
+First, create a workspace on the Che Dashboard and choose Visual Studio Code (desktop) as your editor.
+
+image::/assets/img/use-desktop-visual-studio-code/select-editor.png["Create Workspace"]
+{nbsp} +
+
+Wait for the login page to appear and log in with your Microsoft or GitHub account.
+Click the *Copy & Continue to Microsoft* or *Copy & Continue to GitHub* button to copy the authentication code to the clipboard and to open the authenticating pop-up.
+
+image::/assets/img/use-desktop-visual-studio-code/login-with-provider.png["Login with GitHub account"]
+{nbsp} +
+
+Complete the authentication process, paste in the code from the clipboard, and follow the instructions. On successful authentication, close the pop-up and wait for the tunnel to your workspace to be created.
+
+image::/assets/img/use-desktop-visual-studio-code/open-visual-studio-code.png["Tunnel to the workspace"]
+{nbsp} +
+
+Click *Open in Visual Studio Code desktop* to open your workspace in the desktop version of Visual Studio Code. Confirm opening the desktop application by clicking Open Visual Studio Code - URL Handler.
+
+image::/assets/img/use-desktop-visual-studio-code/confirm-opening-vscode-desktop.png["Open in Visual Studio Code desktop"]
+{nbsp} +
+
+Wait for the local Visual Studio Code and allow the *Remote - Tunnels* extension to open the URI.
+
+[NOTE]
+====
+If Visual Studio Code opens your workspace with a delay, open the *Remote Explorer* view.
+====
+
+== Connecting Visual Studio Code desktop to a running workspace
+
+=== Prepare your local Visual Studio Code.
+
+First, you need to install both link:https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server[Remote - Tunnels] and link:https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-explorer[Remote Explorer] extensions to your local Visual Studio Code. It will allow you to connect to a remote machine via a secure tunnel.
+
+=== Install Visual Studio Code CLI.
+
+Next, you need to install Visual Studio Code CLI in your workspace. Open a terminal and run the following command to install Visual Studio Code CLI in the `/checode` directory.
+
+[source, code]
+----
+curl 'https://code.visualstudio.com/sha/download?build=stable&os=cli-alpine-x64' --location -o /tmp/code.tar.gz && \
+tar -xvzf /tmp/code.tar.gz --directory=/checode
+----
+
+[IMPORTANT]
+====
+If you don't have `gzip` installed in your workspace, you won't be able to install the CLI using the command above.
+
+In this case you need to:
+
+. link:https://code.visualstudio.com/Download[download CLI] for *Linux x64* architecture to your desktop
+
+. unpack and upload the extracted binary to your workspace root
+
+. move the binary to `/checode` directory and make it executable by `mv $PROJECT_SOURCE/code /checode && chmod +x /checode/code`
+====
```

Comment:
@vitaliy-guliy side question, should we create an issue to improve this flow?

Reply:
ok, we probably need to create an issue for that and discuss on the sprint planning next week